### ISSUE TITLE
svunit.f does not use paths relative to the location of the svunit.f file.

### ISSUE THREAD

**Issue by BenjaminGittins** [svunit/svunit#285]
Hi, 

I am having problems with the paths in svunit.f file(s).

Consider the following directory structure.

```
/src/hdl
/src/hdl/svunit.f
/src/hdl/core/lib_01/*.sv
/src/hdl/core/lib_02/*.sv

```

the file of svunit.f has a first line...
```

core/lib_01/first_file.sv

```

The following command prompt works: 
`   runSVUnit `

However, setting the output directory for temporary files breaks the project because svunit can no longer find the files to include.
`runSVUnit -o svunit_results`

More to the point, if i have multiple svunit.f files (e.g. one for a folder containing .svh and .sv files for a specific customer), i would want including two svunit.f files to find the files relative to the location of that respective svunit.f file.

I think the code responsible for the ... (i think) ... incorrect behavior is probably here: 

https://github.com/svunit/svunit/blob/5b785137a7563ac0b145fa660c2529bb33037d9d/bin/runSVUnit#L190C1-L195C1

Thanks.

**Comment by tudortimi** [comment]
Simulators will generally interpret relative paths as being relative to the tool invocation directory. There's not much we can do to change this behavior.

**Comment by tudortimi** [comment]
Actually, it might be a matter of using `-f` (relative to tool invocation directory) or `-F` (relative to args file). This isn't a universal convention. Many tools support this, but probably our favorite Vivado Simulator doesn't.

**Comment by BenjaminGittins** [comment]
Maybe the other option is to parse the svunit.f files, and supply each .sv or .svh file or include directory file with full path names to all simulators. This would ensure consistent behavior no matter where the -o output folder was located, across all simulators (without worrying about simulator specific -f or -F options). It would also make it easier to include multiple svunit.f files from different locations when working with large multiple library testing.

**Comment by BenjaminGittins** [comment]
Here is an example of relative paths in SVUnit causing problems in Questa:

-- Compiling module sll_phy_t01_apb3t_t01_ro_t01_fsm
** Warning: ** while parsing file included at /tools/prj_sll/sll_ip_repo/sll_cores/sll_xspi_mbmc3_3_p_p1i1x1_v_1.0/hdl/sll_core_vivado_tb/./../../../sll_phy1_iob1_xspi1_apb3t1_v_1.4/hdl/sll_phy_t01_svunit_tb/sll_phy_t01_apb3t_t01/sll_phy_t01_apb3t_t01_ro_t01_fsm_tb01_unit_test.sv(24)
** at ../../../../sll_phy1_iob1_xspi1_apb3t1_v_1.4/hdl/sll_phy_t01/sll_phy_t01_apb3t_t01/sll_phy_t01_apb3t_t01_ro_t01_fsm.sv(19): (vlog-2275) Existing module 'sll_phy_t01_apb3t_t01_ro_t01_fsm' at line 19 will be overwritten.

I am fairly confident SVUnit is (in one way or another) importing the same sll_phy_t01_apb3t_t01_ro_t01_fsm.sv file, but because it uses two different relative paths are provided to the same file, Questa can get upset (maybe its doing simple string compare of the path/filename to check if same/different.  This doesn't happen every time (i think incremental compilation of simulation files removes this error), but i shouldn't have to run the test twice to avoid error messages from the simulator. 

Therefore, I think its a really good idea to ensure that the simulation tools receive all filenames from SVUnit with absolute paths.  Which means parsing the svunit.f files to convert relative paths to absolute paths.

**Comment by tudortimi** [comment]
@BenjaminGittins I envision reducing the code in the `runSVUnit`/`buidSVUnit` scripts to a minimum. They currently handle way too much, because they have to also handle not just SVUnit aspects, but also build management (i.e. compile the code under test, simulator arguments for different simulators, etc.).

I have another project for the build management aspects, which can interact with SVUnit: https://github.com/tudortimi/gradle-hdvl. That's supposed to handle the compilation aspects and generate args files for use in, among others, SVUnit (i.e. generate the `svunit.f` file). It's pretty rough at the moment, because it's missing documentation and lot of things still need to be worked out, but it has the potential to improve developer experience significantly.

**Comment by BenjaminGittins** [comment]
Yes, i was reluctant to suggest a refactoring of the code... but if your doing that, it could significantly improve management of the ever-increasing tool support variants. 

For this moment, the work i've done in the fork helps get Xilinx complaining less / more usable with the current SVUnit implementation (let me know if/how to help you bring that back into your main code base).  Once you get the gradle-hdvl up to beta, please let me know, as that would seem by far the best way forward.
